### ISSUE TITLE
Icon: Critical background icons disappear in windows high contrast mode (WHCM)

### ISSUE THREAD

**Issue by ianmcburnie** [eBay/skin#233]
By default, Windows hides background images in high contrast mode (EDIT: windows no longer hides background images in Edge. See latest comments below for details). CSS allows us to override this using `-ms-high-contrast-adjust: none;` or to provide custom high-contrast mode icons via the `-ms-high-contrast` media query.

Not all background-icons need the override, only critical ones or those that give affordance.

![screenshot 9](https://user-images.githubusercontent.com/38065/41736371-1b9b0d32-7541-11e8-888d-4a82ed3e5137.png)

![screenshot 10](https://user-images.githubusercontent.com/38065/41736394-2ad83536-7541-11e8-8483-c0a7affe774d.png)
![screenshot 12](https://user-images.githubusercontent.com/38065/41736464-72aa0204-7541-11e8-8b46-25cae1c414c2.png)
![screenshot 13](https://user-images.githubusercontent.com/38065/41736465-72d1d126-7541-11e8-95a0-3d7466ac32e7.png)
![screenshot 14](https://user-images.githubusercontent.com/38065/41736466-730b88c6-7541-11e8-8430-dffb07fb42fe.png)

![screenshot 11](https://user-images.githubusercontent.com/38065/41736395-2afcf6e6-7541-11e8-91c7-5dd3aefb0e16.png)

![screenshot 18](https://user-images.githubusercontent.com/38065/41736481-7f935a10-7541-11e8-8dea-82ae7b644d26.png)

**Comment by senthilp** [comment]
A [Twitter thread](https://twitter.com/HugoGiraudel/status/877123359418576896) on Microsoft high contrast mode. Looks like overriding high contrast mode may [not help](https://www.smashingmagazine.com/2018/06/placeholder-attribute/#high-contrast-mode) all people.

**Comment by ianmcburnie** [comment]
@senthilp that thread is only related to the placeholder attribute in high contrast mode, and how overriding it might not be a help.

Overriding it for critical icons is definitely a help. As would be providing a text alternative.

**Comment by ianmcburnie** [comment]
`Microsoft Edge will no longer remove background images, and instead will render an opaque layer behind text to improve legibility while maintaining the site design.`

Taken from: https://blogs.windows.com/msedgedev/2016/04/20/building-a-more-accessible-web-platform/

So I think what is going on is not that the background images are being completely removed, just that they become illegible due to dark stroke/fill on dark background.

More investigation needed.

**Comment by ianmcburnie** [comment]
One possible solution, if we want to continue using background svg, is to recreate the entire icon set with inverted colours and then use the media query:

```css
@media screen and (-ms-high-contrast: black-on-white) {
  .dialog__close span {
    background-url: white_on_black_base64_encoded_icon;
  }
}
```

**Comment by ianmcburnie** [comment]
I feel like we should fix checkbox and radio ASAP as they are critical icons. Let's create inverted versions of the background SVGs for those for v5.1.0.

**Comment by ianmcburnie** [comment]
Here's an example of the high contrast white theme affecting the white background circle of page notice icons. Less critical of an issue in this case, but shows how we should cater our icons for both black-on-white and white-on-black.

![screenshot 23](https://user-images.githubusercontent.com/38065/41885349-7beaeeb4-78ac-11e8-9041-5d0a47fef69a.png)

**Comment by ianmcburnie** [comment]
As an interim solution for the icon module (until we tackle #239), we could create an `icon-high-contrast.css` dist file (containing the -ms-high-contrast media queries and background-image overrides) and a respective browser.json. Apps using background icons could include this. Apps only using foreground icons would not need to include this.

**Comment by ianmcburnie** [comment]
Specific instances of icons within Skin modules (e.g. dialog, pagination, menu) have been fixed in PR #242.

The icon module itself will be addressed in v6.0.0 and therefore this ticket will remain open to track that.

**Comment by ianmcburnie** [comment]
Can potentially wait till after v6. Doesn't seem urgent for coreui.

**Comment by ianmcburnie** [comment]
Going to close this as we did solve the critical icons that appear in our modules (checkbox, radio, menu, etc), as per the description. I'm going to create a new ticket for us to create light versions of all icons in our background icon set.
